appcues/src/main/java/com/appcues/ui/modal/DialogModal.kt

```diff
@@ -33,48 +43,119 @@ private val MAX_WIDTH_EXPANDED_DP = 560.dp
 private val MAX_HEIGHT_COMPACT_DP = Dp.Unspecified
 private val MAX_HEIGHT_MEDIUM_DP = 800.dp
 private val MAX_HEIGHT_EXPANDED_DP = 900.dp
-private const val SCREEN_PADDING = 0.05
+
+private val SCREEN_HORIZONTAL_PADDING = 12.dp
+private val SCREEN_VERTICAL_PADDING = 24.dp
+
+internal data class DialogTransition(
+    val type: String,
+    val slideInEdge: String?,
+    val slideOutEdge: String?,
+)
+
+internal enum class TransitionEdge {
+    LEADING, TRAILING, TOP, BOTTOM, CENTER
+}
 
 @Composable
 internal fun DialogModal(
     style: ComponentStyle?,
     content: @Composable (modifier: Modifier, containerPadding: PaddingValues, safeAreaInsets: PaddingValues) -> Unit,
-    windowInfo: AppcuesWindowInfo
+    windowInfo: AppcuesWindowInfo,
+    transition: DialogTransition,
 ) {
-    val configuration = LocalConfiguration.current
-    val dialogHorizontalMargin = (configuration.screenWidthDp * SCREEN_PADDING).dp
-    val dialogVerticalMargin = (configuration.screenHeightDp * SCREEN_PADDING).dp
     val isDark = isSystemInDarkTheme()
+    val density = LocalDensity.current
 
     val maxWidth = maxWidthDerivedOf(windowInfo)
     val maxHeight = maxHeightDerivedOf(windowInfo)
 
-    AppcuesContentAnimatedVisibility(
-        enter = dialogEnterTransition(),
-        exit = dialogExitTransition(),
+    val horizontalPaddingPx = with(density) { SCREEN_HORIZONTAL_PADDING.roundToPx() }
+    val verticalPaddingPx = with(density) { SCREEN_VERTICAL_PADDING.roundToPx() }
+
+    Box(
+        modifier = Modifier
+            .fillMaxSize()
+            .padding(horizontal = SCREEN_HORIZONTAL_PADDING, vertical = SCREEN_VERTICAL_PADDING)
     ) {
-        Box(modifier = Modifier.fillMaxSize()) {
+        AppcuesContentAnimatedVisibility(
+            modifier = Modifier.align(style.getBoxAlignment()),
+            enter = transition.toEnterTransition(style, horizontalPaddingPx, verticalPaddingPx),
+            exit = transition.toExitTransition(style, horizontalPaddingPx, verticalPaddingPx),
+        ) {
```

Comment:
The animation needs to go here - inside the `Box`. This is because we don't want to animate the entire box (mostly whitespace around the dialog), but rather, the modal contents. This doesn't matter much on `fade` animations, but on `slide` animations its the difference between sliding the entire screen in from an edge (undesired) versus sliding just the content box in from the edge (desired)